aten/src/TH/generic/THTensor.cpp

```diff
@@ -478,17 +478,29 @@ void THTensor_(unsqueeze1d)(THTensor *self, THTensor *src, int dimension)
 
   THTensor_(set)(self, src);
 
-  self->resize_dim(self->dim() + 1);
-  for (d = self->dim()-1; d > dimension; d--) {
-    self->set_size(d, self->size(d-1));
-    self->set_stride(d, self->stride(d-1));
+  std::vector<int64_t> newSize(/* size */ self->dim()+1);
+  std::vector<int64_t> newStride(/* size */ self->dim()+1);
+
+  for(d = self->dim(); d > dimension; d--)
```

Comment:
This doesn't look semantics preserving

Reply:
Specifically, we previously began the loop on d at `dim()-1`; and now you are beginning the loop at `dim()`

Reply:
Previously, we began the loop at self->dim()-1 because self->dim() was expanded before this loop was run. Now that the operation is no longer performed in-place, this is unnecessary. Therefore, we now start at self->dim().

Reply:
You are right!